site/app/views/forum/ForumThreadView.php

```diff
@@ -47,7 +47,10 @@ function openFile(directory, file, path ){
 
 			$( document ).ready(function() {
 			    enableTabsInTextArea('post_content');
-			    saveScrollLocationOnRefresh('thread_list');
+				saveScrollLocationOnRefresh('thread_list');
+				var previous_scroll_position = localStorage.getItem("scroll_position");
```

Comment:
We should be modifying saveScrollLocationOnRefresh in order to be able to save multiple scroll bar positions in sessionStorage, take a look at this [https://stackoverflow.com/a/5523174](https://stackoverflow.com/a/5523174), it explains the difference between localStorage and sessionStorage.

Within the function you can do something similar to: `sessionStorage.setItem(id + '_scrollTop', $(element).scrollTop());` to set the unique scroll position. Where `id` is the div id, so the first line should also be changed to use the div id not the class name. Then modify the set line to retrieve the correct value from sessionStorage. As a result of this, `resetScrollPosition()` will have to be modified to clear the saved locations on a redirect, and called in the appropriate location.